index.js

```diff
@@ -212,6 +212,12 @@ module.exports = {
         */
         'object-shorthand': ['error', 'properties', { avoidQuotes: true }],
 
+        /*
+            Destructuring should be considered for multi-variable declarations and assignments while simple single
+            variable declarations and assignments likely don't require it.
+        */
+        'prefer-destructuring': 'off',
```

Comment:
I don't see a built-in rule to enforce it but I'd go a step further and say in the comment we should discourage [destructuring with renaming](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Assigning_to_new_variable_names).

Reply:
Why do you think we shouldn't allow this?

Reply:
Renaming is fine and often required.

Reply:
I find the syntax of destructuring with renaming to not be that much less verbose but I do find it less clear (I can scan the second one and understand the outcome quicker). If I ask myself, what are the properties of o that this depends on, it's more readily apparent in the second example.

```js
const {a: awesomeA, b: awesomeB} = o;
```
vs
```js
const awesomeA = o.a;
const awesomeB = o.b;
```

I find visually parsing it more difficult than without destructuring, particularly with destructuring in function parameters:
```js
function test (something, {a: awesomeA, b: awesomeB} = config) {...}
```
vs
```js
function test (something, config) {
    const awesomeA = config.a;
    const awesomeB = config.b;
}
```

Particularly if you add defaults and have more than two properties it becomes a visual soup of brackets and things that look like object literals. The syntax for destructuring with renaming is so much like the object literal syntax that I find it a code scanning speedbump. It's the same reason I advocated for TypeScript interface definition over the type alias format (where type aliases look a lot like object literals).

But I also don't feel strongly about it. I'll align with the majority vote. Do @jattasNI or @LukasKall have an opinion?

Reply:
I agree that at a certain point, destructuring with renaming becomes very complex and less clear. But I don't feel like we should discourage destructuring with renaming at all. It's a personal choice at which point it's to complex and in some cases its quite useful, e.g. when destructuring multiple properties and only renaming one.

```javascript
const { a, b, c: betterC } = obj;
```

Reply:
I haven't seen destructuring used much since it's a new-ish feature so all of the syntaxes are harder to read for me than splitting it into separate assignments. But the more I see the syntax (including the rename version), the more I like it so I don't think we should discourage it.

Reply:
sounds good to me! thanks for the discussions 👍